R/kiss-reports.R

```diff
@@ -4,10 +4,53 @@
 #'
 #'  This is a constructor function.
 #'
+#' @return If url is a valid string or url (from \code{httr::url}) this returns
+#'   a new KissReports object that can be passed to the read S3 generic to
+#'   access all reports available in your configured account
 #'  @export
 KissReports <- function() {
   url <- httr::parse_url("https://api.kissmetrics.com/core/reports")
+
   structure(list(
     url = url),
     class = "KissReports")
 }
+
+
+#' Read a list of reports from a KissReports object
+#'
+#' This function reads all the data from a KissMetrics people report
+#' returning everything in a data frame. Please note this is a synchronous
+#' function that can take quite some time to complete.
+#'
+#' @return  A \code{data.frame} containing report names and urls
+#' @export
+read.KissReports <- function(reports) {
+  # Make request
+  headers <- c(authorizationHeader(), jsonHeader())
+  body <- ""
+  encoding <- "json"
+
+  requestKey <- c(reports$url, headers, body, encoding)
+  response <- readCache(reports, requestKey)
+  if (is.null(response)) {
+    response <- httr::GET(reports$url,
+                           encode = encoding,
+                           httr::add_headers(.headers = headers))
+
+    httr::stop_for_status(response)
+    writeCache(reports, requestKey, response)
+  }
+
+  # Get list of reports
+  links  <- jsonlite::fromJSON(httr::content(response, "text"))$links
+  statusLink <- links[links$name == "Status", "href"]
+
+  reportsLink <- with(result, links[links$name == "First", "href"])
+  print("Getting all reports")
+  results <- as.data.frame(loadPages(reports, reportsLink), stringsAsFactors = FALSE)
+  urlExtractor <- function(df) { df[3,3] }
```

Comment:
You should be explicit here about what grabbing the (3,3) entry is doing -- will this always be the entry you want? Why?